contrib/generate-api-changes.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+#
+# Creates a temporary branch with an api-changes patch for each commit back to master.
+
+set -euo pipefail
+
+REPO_DIR=$(git rev-parse --show-toplevel)
+API_DIR="$REPO_DIR/api"
+
+NIGHTLY=$(cat nightly-version)
+# Our docs have broken intra doc links if all features are not enabled.
+RUSTDOCFLAGS="-A rustdoc::broken_intra_doc_links"
+
+# `sort -n -u` doesn't work for some reason.
+SORT="sort --numeric-sort"
+
+# Get temporary branch name.
+TEMP_BRANCH=$(mktemp -u temp-branch-XXXXXX)
+
+# Sort order is effected by locale. See `man sort`.
+# > Set LC_ALL=C to get the traditional sort order that uses native byte values.
+export LC_ALL=C
+
+main() {
+    need_nightly
+
+    # Create a new temporary branch
+    current_branch=$(git branch --show-current)
+
+    # Find commits on current branch that are not in master
+    commits=$(git rev-list master.."$current_branch")
+
+    git checkout -b "$TEMP_BRANCH" master || exit 1
```

Comment:
In f84fb80c6d894b49c2f1d909bf6ad865a908bcd2:

This `exit 1` doesn't do anything except throwing away the return value from `git`, since you have `set -e` on the script.